src/com/amplitude/api/DatabaseHelper.java

```diff
@@ -321,6 +321,12 @@ protected synchronized Object getValueFromTable(String table, String key) {
                 String.format("DB: Failed to getValue: %s", key), e
             );
             delete();
+        } catch (IllegalStateException e) {  // put before Runtime since IllegalState extends
```

Comment:
Maybe do something slightly more specific? Like check that the message refers to `CursorWindow`.